### PR TITLE
Add talk to filter query docs

### PR DESCRIPTION
**Pull request by KnorpelSenf**
As requested in https://t.me/grammyjs/247463

### CODE REVIEW

File: site/docs/guide/filter-queries.md [hunk 1]
```diff
@@ -347,8 +347,24 @@ In this case, you will be provided with helpful error messages.
...
+If a property is inferred to be present, you can safely rely on it.
+If a property is inferred to be potentially absent, then this means that there are certain cases of it missing.
+It is not a good idea to perform type casts with the `!` operator.
+
+> It may not be obvious to you what those cases are.
+> Don't hesitate to ask in the [group chat](https://t.me/grammyjs) if you cannot figure it out.
+
+Computing these types is complicated.
+A lot of knowledge about the Bot API went into this part of grammY.
+If you want to understand more about the basic approaches to how these types are computed, there is a [talk on YouTube] that you can watch.
```
Comment (roj1512) [inline site/docs/guide/filter-queries.md:370]: Missed link

Comment (KnorpelSenf) [reply site/docs/guide/filter-queries.md:370]: oof

Comment (KnorpelSenf) [inline site/docs/guide/filter-queries.md:370]:
```suggestion
If you want to understand more about the basic approaches to how these types are computed, there is a [talk on YouTube](https://youtu.be/ZvT_xexjnMk) that you can watch.
```

File: site/docs/id/guide/filter-queries.md [hunk 2]
```diff
@@ -350,8 +350,23 @@ Selain itu, kamu juga akan diberikan pesan error yang lebih bermanfaat.
...
+Jika suatu properti type infer-nya tersedia, kamu dapat dapat menggunakan type tersebut dengan aman.
+Jika suatu properti type infer-nya hilang, berarti ada kasus tertentu yang membuat type-nya hilang.
+Bukan ide yang baik untuk melakukan casting type dengan operator `!`.
+
+> Mungkin ini kurang jelas tentang apa maksud kasus tersebut.
+> Jangan ragu untuk bertanya di [obrolan grup](https://t.me/grammyjs) jika kamu tidak memahaminya.
+
+Membuat type ini merumitkan.
+Banyak sekali pengetahuan tentang API Bot yang masuk pada bagian grammY ini.
+Apabila kamu ingin lebih memahami tentang pendekatan dasar bagaimana type ini dibuat, berikut ada [pembahasannya di Youtube](https://youtu.be/ZvT_xexjnMk) yang mungkin bisa kamu tonton.
```
Comment (KnorpelSenf) [inline site/docs/id/guide/filter-queries.md:372]: @dzikrl perhaps it should be mentioned that the presentation on YouTube is in English (as a warning), what do you think?

File: site/docs/uk/guide/filter-queries.md [hunk 3]
```diff
@@ -347,8 +347,24 @@ bot.on(":text", handler);
...
+Якщо властивість виводиться як присутня, ви можете безпечно покладатися на неї.
+Якщо властивість виводиться як потенційно відсутня, то це означає, що існують певні випадки, коли вона відсутня.
+Не варто виконувати приведення типів за допомогою оператора `!`.
+
+> Для вас може бути неочевидним, що це за випадки.
+> Не соромтеся запитувати в [груповому чаті](https://t.me/grammyjs), якщо не можете розібратися.
+
+Обчислення цих типів є складним.
+При розробці запитів фільтрування було використано багато знань про Bot API.
+Якщо ви хочете дізнатися більше про основні підходи до обчислення цих типів, ви можете подивитися [виступ на YouTube](https://youtu.be/ZvT_xexjnMk).
```
Comment (KnorpelSenf) [inline site/docs/uk/guide/filter-queries.md:370]: @niusia-ua perhaps it should be mentioned that the presentation on YouTube is in English (as a warning), what do you think?

Comment (niusia-ua) [reply site/docs/uk/guide/filter-queries.md:370]:
```suggestion
Якщо ви хочете дізнатися більше про основні підходи до обчислення цих типів, ви можете подивитися [виступ на YouTube](https://youtu.be/ZvT_xexjnMk).

::: tip
Зауважте, що ця доповідь викладається англійською мовою.
Якщо ви погано розумієте розмовну англійську, скористайтеся англійськими або автоперекладеними субтитрами.
:::
```

File: site/docs/uk/guide/filter-queries.md [hunk 4]
```diff
@@ -347,8 +347,24 @@ bot.on(":text", handler);
 Перевірка запитів фільтрування відбувається лише один раз, коли бот ініціалізується та викликається `bot.on()`.
 
 Під час запуску grammY створює предикатну функцію із запиту фільтрування, розділяючи його на частини.
-Кожна частина буде зіставлена з функцією, яка виконує одну перевірку `in` або дві перевірки, якщо частина запиту пропущена і потрібно перевірити два значення.
+Кожна частина буде зіставлена з функцією, яка виконує одну перевірку одну перевірку істинності для властивості об'єкта або дві перевірки, якщо частина запиту пропущена і потрібно перевірити два значення.
```
Comment (LWJerri) [inline site/docs/uk/guide/filter-queries.md:350]:
```suggestion
Кожна частина буде зіставлена з функцією, яка виконує одну перевірку істинності для властивості об'єкта або дві перевірки, якщо частина запиту пропущена і потрібно перевірити два значення.
```

**Review by LWJerri — CHANGES_REQUESTED** [review]
not :shipit:, because I leave 1 comment.

File: site/docs/uk/guide/filter-queries.md (hunk 3 above)
Comment (KnorpelSenf) [reply site/docs/uk/guide/filter-queries.md:370]: That is a really good point. I should add subtitles to my video. We can also translate them to other languages.

File: site/docs/id/guide/filter-queries.md (hunk 2 above)
Comment (dzikrl) [reply site/docs/id/guide/filter-queries.md:372]: > @dzikrl perhaps it should be mentioned that the presentation on YouTube is in English (as a warning), what do you think?

It's okay to not mention it. In [grammY introduction](https://grammy.dev/id/guide/introduction#persiapan-untuk-memulai) about "Not Learning How to Code" on Indonesian translation. There is also a link Youtube video that in English too. But, @quadratz improve that with a some note.

cc : @quadratz?

File: site/docs/uk/guide/filter-queries.md (hunk 3 above)
Comment (niusia-ua) [reply site/docs/uk/guide/filter-queries.md:370]: I don't think this is really necessary. YouTube's auto translated subtitles are good enough, so spectators can understand the main idea.

File: site/docs/id/guide/filter-queries.md (hunk 2 above)
Comment (dzikrl) [inline site/docs/id/guide/filter-queries.md:372]:
```suggestion
Apabila kamu ingin lebih memahami tentang pendekatan dasar bagaimana type ini dibuat, berikut ada [pembahasannya di Youtube](https://youtu.be/ZvT_xexjnMk) yang mungkin bisa kamu tonton.

::: tip
Catatan penerjemah :

Pembahasan pada video di atas dalam berbahasa Inggris.
Apabila kamu merasa kurang jelas/kesulitan dalam memahami penjelasan tersebut, gunakanlah fitur subtitle dan terjemahan otomatisnya.
:::
```

**Review by dzikrl — CHANGES_REQUESTED** [review]
cc : @quadratz

**Review by LWJerri — APPROVED** [review]
:shipit:

File: site/docs/id/guide/filter-queries.md [hunk 5]
```diff
@@ -29,7 +29,7 @@ bot.on("message:text", async (ctx) => {
 });
 ```
 
-Filter query grammY diimplementasikan baik di _runtime_ maupun di _type level_.
+Filter query grammY diimplementasikan baik di [_runtime_](#performa) maupun di [_type level_](#type-safety).
```
Comment (quadratz) [inline site/docs/id/guide/filter-queries.md:32]:
```suggestion
Filter query grammY tersedia di level [_runtime_](#performa) dan [_type_](#type-safety).
```

File: site/docs/id/guide/filter-queries.md [hunk 6]
```diff
@@ -350,8 +350,23 @@ Selain itu, kamu juga akan diberikan pesan error yang lebih bermanfaat.
 Validasi filter query hanya dilakukan sekali, ketika bot diinisialisasi dan `bot.on()` dipanggil.
 
 Ketika dimulai, grammY menurunkan _function predicate_ dari filter query dengan cara memecahnya menjadi beberapa komponen query.
-Setiap komponen akan di-map ke sebuah function yang mengerjakan satu pemeriksaan `in`, atau dua pemeriksaan jika komponen tersebut diabaikan dan dua value perlu dilakukan pemeriksaan.
+Setiap komponen akan di-map ke sebuah function yang mengerjakan satu pemeriksaan truthiness untuk properti objek, atau dua pemeriksaan jika komponen tersebut diabaikan dan dua value perlu dilakukan pemeriksaan.
```
Comment (quadratz) [inline site/docs/id/guide/filter-queries.md:353]:
```suggestion
Setiap komponen akan dipetakan ke sebuah function untuk diperiksa apakah properti objek tersebut cocok dengan filter terkait, atau bahkan dua kali pemeriksaan jika komponen tersebut diabaikan dan dua nilai perlu diperiksa.
```

File: site/docs/guide/filter-queries.md [hunk 7]
```diff
@@ -347,8 +347,24 @@ In this case, you will be provided with helpful error messages.
 The validation of the filter queries happens only once, when the bot is initialized and `bot.on()` is called.
 
 On start-up, grammY derives a predicate function from the filter query by splitting it into its query parts.
-Every part will be mapped to a function that performs a single `in` check, or two checks if the part is omitted and two values need to be checked.
+Every part will be mapped to a function that performs a single truthiness check for an object property, or two checks if the part is omitted and two values need to be checked.
```
Comment (quadratz) [inline site/docs/guide/filter-queries.md:350]: @KnorpelSenf Could you provide an example for the scenario mentioned in this line `two checks if the part is omitted and two values need to be checked`? I'm a bit confused about when the part is omitted.

Comment (KnorpelSenf) [reply site/docs/guide/filter-queries.md:350]: `":text"` will be expanded to `["message:text", "channel_post:text"]` so the omitted value before the colon requires two properties to be checked.

File: site/docs/id/guide/filter-queries.md [hunk 8]
```diff
@@ -350,8 +350,23 @@ Selain itu, kamu juga akan diberikan pesan error yang lebih bermanfaat.
...
+Secara umum, kamu dapat mempercayai type inference itu ia selalu bekerja dengan benar.
+Jika suatu properti type infer-nya tersedia, kamu dapat dapat menggunakan type tersebut dengan aman.
+Jika suatu properti type infer-nya hilang, berarti ada kasus tertentu yang membuat type-nya hilang.
+Bukan ide yang baik untuk melakukan casting type dengan operator `!`.
+
+> Mungkin ini kurang jelas tentang apa maksud kasus tersebut.
+> Jangan ragu untuk bertanya di [obrolan grup](https://t.me/grammyjs) jika kamu tidak memahaminya.
+
+Membuat type ini merumitkan.
+Banyak sekali pengetahuan tentang API Bot yang masuk pada bagian grammY ini.
```
Comment (quadratz) [inline site/docs/id/guide/filter-queries.md:371]:
```suggestion
Umumnya, kamu bisa mempercayai hasil penyeleksian tersebut benar-benar terseleksi dengan baik.
Contohnya, jika suatu property tersedia, kamu dapat menggunakan type tersebut dengan aman.
Namun, jika suatu property tidak tersedia, berarti ada proses tertentu yang mengakibatkan type property tersebut tidak terseleksi.
Kami tidak menyarankan untuk mengakalinya dengan melakukan type casting menggunakan operator `!`.

> Meski terlihat jelas, dalam kasus tertentu, mungkin kamu tidak menemukan penyebab suatu property tidak terseleksi dengan benar.
> Jangan ragu untuk bertanya di [obrolan grup](https://t.me/grammyjs) jika kamu masih belum menemukan penyebabnya.

Mengelola berbagai macam type berikut bukanlah perkara mudah.
Banyak sekali pengetahuan tentang API Bot yang masuk pada bagian grammY ini.
```

File: site/docs/id/guide/filter-queries.md (hunk 2 above)
Comment (quadratz) [reply site/docs/id/guide/filter-queries.md:372]:
```suggestion
Apabila kamu ingin memahami lebih dalam bagaimana type ini dibuat, berikut pembahasan lengkapnya yang bisa kamu [tonton di Youtube](https://youtu.be/ZvT_xexjnMk).
```

Comment (quadratz) [reply site/docs/id/guide/filter-queries.md:372]: I don't see the Indonesian subtitle available for that video (yet), so it's not relevant to suggest the reader to turn on the subtitle.

**Review by quadratz — CHANGES_REQUESTED** [review]

File: site/docs/guide/filter-queries.md (hunk 7 above)
Comment (quadratz) [reply site/docs/guide/filter-queries.md:350]: Right. That makes sense now. Thanks.

File: site/docs/id/guide/filter-queries.md (hunk 2 above)
Comment (KnorpelSenf) [reply site/docs/id/guide/filter-queries.md:372]: It should be
```suggestion
Apabila kamu ingin lebih memahami tentang pendekatan dasar bagaimana type ini dibuat, berikut ada [pembahasannya di YouTube](https://youtu.be/ZvT_xexjnMk) yang mungkin bisa kamu tonton.
```
right?

Comment (KnorpelSenf) [reply site/docs/id/guide/filter-queries.md:372]: I'm working on the subtitles

Comment (quadratz) [reply site/docs/id/guide/filter-queries.md:372]: > I'm working on the subtitles

Okay. We will leave the tips then. Do you need help to generate ID subtitle as well?

Comment (KnorpelSenf) [reply site/docs/id/guide/filter-queries.md:372]: I think that @niusia-ua is right in https://github.com/grammyjs/website/pull/1074#discussion_r1642602223 so it isn't needed, but if the translation is bad, you can contribute it later

File: site/docs/id/guide/filter-queries.md [hunk 9]
```diff
@@ -350,8 +350,23 @@ Selain itu, kamu juga akan diberikan pesan error yang lebih bermanfaat.
...
-Setiap komponen akan di-map ke sebuah function yang mengerjakan satu pemeriksaan `in`, atau dua pemeriksaan jika komponen tersebut diabaikan dan dua value perlu dilakukan pemeriksaan.
+Setiap komponen akan di-map ke sebuah function yang mengerjakan satu pemeriksaan truthiness untuk properti objek, atau dua pemeriksaan jika komponen tersebut diabaikan dan dua value perlu dilakukan pemeriksaan.
 Function-function ini kemudian disatukan untuk membentuk sebuah predicate yang akan memeriksa sebanyak mungkin value yang relevan untuk query, tanpa melakukan proses perulangan terhadap key object `Update`.
 
 Sistem ini menggunakan lebih sedikit operasi dibandingkan dengan beberapa library lainnya, dimana dibutuhkan beberapa pengecekan array ketika melakukan routing update.
 Ini membuat sistem filter query grammY selain lebih unggul juga jauh lebih cepat.
+
+### Type Safety
+
+Sebagaimana yang disebutkan diatas, filter queries secara otomatis akan mempersempit properti tertentu pada objek context dan menghasilkan satu atau lebih predicate dalam fungsi predicate type di TypeScript.
```
[the remark this answers is not in the document]
Comment (dzikrl) [reply site/docs/id/guide/filter-queries.md:361]: This. I don't know how to write it well.

**Review by dzikrl — APPROVED** [review]

File: site/docs/id/guide/filter-queries.md [hunk 10]
```diff
@@ -350,8 +350,24 @@ Selain itu, kamu juga akan diberikan pesan error yang lebih bermanfaat.
...
+Contohnya, jika suatu property tersedia, kamu dapat menggunakan type tersebut dengan aman.
+Namun, jika suatu property tidak tersedia, berarti ada proses tertentu yang mengakibatkan type property tersebut tidak terseleksi.
+Kami tidak menyarankan untuk mengakalinya dengan melakukan type casting menggunakan operator `!`.
+
+> Meski terlihat jelas, dalam kasus tertentu, mungkin kamu tidak menemukan penyebab suatu property tidak terseleksi dengan benar.
+> Jangan ragu untuk bertanya di [obrolan grup](https://t.me/grammyjs) jika kamu masih belum menemukan penyebabnya.
+
+Mengelola berbagai macam type berikut bukanlah perkara mudah.
+Banyak sekali pengetahuan tentang API Bot yang masuk pada bagian grammY ini.
+Apabila kamu ingin memahami lebih dalam bagaimana type ini dibuat, berikut pembahasan lengkapnya yang bisa kamu [tonton di YouTube](https://youtu.be/ZvT_xexjnMk)(tersedia subtitle bahasa Indonesia).
```
Comment (quadratz) [inline site/docs/id/guide/filter-queries.md:373]: Based on https://github.com/grammyjs/website/pull/1074#discussion_r1646137445, this note is no longer valid.

```suggestion
Apabila kamu ingin memahami lebih dalam bagaimana type ini dibuat, berikut pembahasan lengkapnya yang bisa kamu [tonton di YouTube](https://youtu.be/ZvT_xexjnMk).
```

**Review by quadratz — CHANGES_REQUESTED** [review]

**Review by quadratz — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**